### PR TITLE
Updates needed to get aux_clm tests working with CMEPS

### PR DESCRIPTION
**Pull request by mvertens**
These are updates that were needed to make progress in getting the aux_clm test suite to work with CMEPS.

Test suite: scripts_regression_tests.py 
   there are still outstanding issues to have the scripts_regression_tests complete successfully with CMEPS
Test baseline: NA
Test namelist changes: NA
Test status: bit for bit
Fixes: None
User interface changes?:  None
Update gh-pages html (Y/N)?: N
Code review:

### CODE REVIEW

File: src/components/data_comps/datm/nuopc/atm_comp_nuopc.F90 [hunk 1]
```diff
@@ -409,8 +421,62 @@ subroutine InitializeRealize(gcomp, importState, exportState, clock, rc)
     call NUOPC_CompAttributeGet(gcomp, name='mesh_atm', value=cvalue, rc=rc)
     if (ChkErr(rc,__LINE__,u_FILE_u)) return
 
-    Emesh = ESMF_MeshCreate(filename=trim(cvalue), fileformat=ESMF_FILEFORMAT_ESMFMESH, rc=rc)
-    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    if (cvalue == 'create_mesh') then
```
Comment (jedwards4b) [inline src/components/data_comps/datm/nuopc/atm_comp_nuopc.F90]: Wouldn't this be more appropriate in share?

**Comment by mvertens** [comment]
At this point this is only being used for regional cases. We could think of
moving it to share - but I am hesitant since we want the nuopc caps to be
as self contained as possible with the idea that the components could
theoretically be used in other systems. Do you want to hold up this PR if
we decide to move this to a share directory - since this will impact both
CLM and DATM.

**Comment by jedwards4b** [comment]
No - just fix this issue:
scripts/lib/CIME/build.py:148:56: W0613: Unused argument 'use_esmf_lib' (unused-argument)

**Comment by mvertens** [comment]
@jedwards4b - thanks I fixed that issue and pushed back to github

**Comment by jedwards4b** [comment]
@mvertens still failing

**Pull request merged**